src/main/java/hobbylist/logic/commands/EditCommand.java

```diff
@@ -106,9 +106,9 @@ private static Activity createEditedActivity(Activity activityToEdit,
         Description updatedDescription = editActivityDescriptor.getDescription()
                 .orElse(activityToEdit.getDescription());
         Set<Tag> updatedTags = editActivityDescriptor.getTags().orElse(activityToEdit.getTags());
-        Optional<Date> date = Optional.empty();
-        if (editActivityDescriptor.getDate().isPresent()) {
-            date = editActivityDescriptor.getDate();
+        Optional<Date> date = null;
+        if (editActivityDescriptor.checkDate() != null) {
```

Comment:
I'm just wondering what was wrong with the previous `editActivityDescriptor.getDate().isPresent()` check.

Reply:
I see